lib/bolt/config.rb

```diff
@@ -107,17 +102,37 @@ def self.default
 
     def self.from_boltdir(boltdir, overrides = {})
       data = Bolt::Util.read_config_file(nil, [boltdir.config_file], 'config') || {}
-      new(boltdir, data, overrides)
+      new(boltdir, data, overrides, load_defaults)
     end
 
     def self.from_file(configfile, overrides = {})
       boltdir = Bolt::Boltdir.new(Pathname.new(configfile).expand_path.dirname)
       data = Bolt::Util.read_config_file(configfile, [], 'config') || {}
+      new(boltdir, data, overrides, load_defaults)
+    end
+
+    def self.load_defaults
+      # Lazy-load expensive gem code
+      require 'win32/dir' if Bolt::Util.windows?
+
+      system_path = if Bolt::Util.windows?
+                      Pathname.new(File.join(Dir::COMMON_APPDATA, 'PuppetLabs', 'bolt', 'etc', 'bolt.yaml'))
+                    else
+                      Pathname.new(File.join('/etc', 'puppetlabs', 'bolt', 'bolt.yaml'))
+                    end
+      user_path = Pathname.new(File.join('~', '.puppetlabs', 'etc', 'bolt', 'bolt.yaml'))
+
+      system_data = Bolt::Util.read_config_file(nil, [system_path], 'config') || {}
+      user_data = Bolt::Util.read_config_file(nil, [user_path], 'config') || {}
 
-      new(boltdir, data, overrides)
+      data = []
+      data << { filepath: system_path, data: system_data } unless system_data.empty?
+      data << { filepath: user_path, data: user_data } unless user_data.empty?
+
+      data
     end
 
-    def initialize(boltdir, config_data, overrides = {})
+    def initialize(boltdir, config_data, overrides = {}, defaults = [])
```

Comment:
It might make more sense to allow `config_data` to be either a hash or an array of hashes and apply the defaults that way. It feels a bit strange to make one of the config files "special".